### PR TITLE
8298044: Fix hidden but significant trailing whitespace in properties files for langtools code

### PR DESCRIPTION
**Pull request by magicus**
According to [the specification](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/util/Properties.html#load(java.io.Reader)) trailing whitespaces in the values of properties files are (somewhat surprisingly) actually significant.

We have multiple files in the JDK with trailing whitespaces in the values. For most of this files, this is likely incorrect and due to oversight, but in a few cases it might actually be intended (like "The value is: ").

After a discussion in the PR for [JDK-8295729](https://bugs.openjdk.org/browse/JDK-8295729), the consensus was to replace valid trailing spaces with the corresponding unicode sequence, `\u0020`. (And of course remove non-wanted trailing spaces.)

Doing so has a dual benefit:

1) It makes it clear to everyone reading the code that there is a trailing space and it is intended

2) It will allow us to remove all actual trailing space characters, and turn on the corresponding check in jcheck to keep the properties files, just like all other source code files, free of trailing spaces.

Ultimately, the call of whether a trailing space is supposed to be there, or is a bug, lies with the respective component teams owning these files. Thus I have split up the set of properties files with trailing spaces in several groups, to match the JDK teams, and open a JBS issue for each of them. This issue is for code I believe belongs with the langtools team.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [ ] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Integration blockers
&nbsp;⚠️ Too few reviewers with at least role reviewer found (have 0, need at least 1) (failed with the updated jcheck configuration)
&nbsp;⚠️ Whitespace errors (failed with the updated jcheck configuration)

### Issue
 * [JDK-8298044](https://bugs.openjdk.org/browse/JDK-8298044): Fix hidden but significant trailing whitespace in properties files for langtools code

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/11487/head:pull/11487` \
`$ git checkout pull/11487`

Update a local copy of the PR: \
`$ git checkout pull/11487` \
`$ git pull https://git.openjdk.org/jdk.git pull/11487/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11487`

View PR using the GUI difftool: \
`$ git pr show -t 11487`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11487.diff">https://git.openjdk.org/jdk/pull/11487.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/11487#issuecomment-1335522006)

### CODE REVIEW

**Comment by magicus** [comment]
**A note to reviewers:**

This PR has been automatically generated by converting all trailing spaces in key-value lines (ignoring empty lines and comments) in the property files into unicode sequences.

In a way, this is a no-op, converting one representation of the already existing space into another. But I think that most of these instances are likely to be bugs, and should thus be fixed.

I think the easiest way to fix those instances is to use the Github "suggestion" review mechanism: 
If you see a value where the trailing space should be removed, press the blue `+` in the gutter as usual, and then in the comment box press the `+/-` icon (or press `ctrl-G`/`cmd-G`). Now you get a copy of the offending line, and can edit it to remove the trailing unicode sequence. I can then easily accept the suggestion into the PR.

But you can also come with review feedback like "remove trailing spaces for all files in directory XXX" and I'll fix that for you.

**Review by jonathan-gibbons — COMMENTED** [review]
All the files are localized versions of the underlying English files.

There are no changes in any of the underlying English files.  That suggests we should just let these localized versions "catch up" with the normal localization process, unless there is reason to believe that will not happen.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/resources/standard_ja.properties [hunk 1]
```diff
@@ -139,7 +139,7 @@ doclet.Subclasses=\u76F4\u7CFB\u306E\u65E2\u77E5\u306E\u30B5\u30D6\u30AF\u30E9\u
 doclet.Subinterfaces=\u65E2\u77E5\u306E\u30B5\u30D6\u30A4\u30F3\u30BF\u30D5\u30A7\u30FC\u30B9\u306E\u30EA\u30B9\u30C8:
 doclet.Implementing_Classes=\u65E2\u77E5\u306E\u5B9F\u88C5\u30AF\u30E9\u30B9\u306E\u30EA\u30B9\u30C8:
 doclet.Functional_Interface=\u6A5F\u80FD\u30A4\u30F3\u30BF\u30D5\u30A7\u30FC\u30B9
-doclet.Functional_Interface_Message=\u3053\u308C\u306F\u6A5F\u80FD\u30A4\u30F3\u30BF\u30D5\u30A7\u30FC\u30B9\u306A\u306E\u3067\u3001\u30E9\u30E0\u30C0\u5F0F\u307E\u305F\u306F\u30E1\u30BD\u30C3\u30C9\u53C2\u7167\u306E\u5272\u5F53\u3066\u30BF\u30FC\u30B2\u30C3\u30C8\u3068\u3057\u3066\u4F7F\u7528\u3067\u304D\u307E\u3059\u3002 
+doclet.Functional_Interface_Message=\u3053\u308C\u306F\u6A5F\u80FD\u30A4\u30F3\u30BF\u30D5\u30A7\u30FC\u30B9\u306A\u306E\u3067\u3001\u30E9\u30E0\u30C0\u5F0F\u307E\u305F\u306F\u30E1\u30BD\u30C3\u30C9\u53C2\u7167\u306E\u5272\u5F53\u3066\u30BF\u30FC\u30B2\u30C3\u30C8\u3068\u3057\u3066\u4F7F\u7528\u3067\u304D\u307E\u3059\u3002\u0020
```
Comment (justin-curtis-lu) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/resources/standard_ja.properties:142]: standard.properties (en, de, zh) does not have a trailing white space for doclet.Functional_Interface_Message.

For consistency, there should not be a space here.

File: src/jdk.jshell/share/classes/jdk/internal/jshell/tool/resources/l10n_de.properties [hunk 2]
```diff
@@ -167,7 +167,7 @@ jshell.console.completion.all.completions.number = <Erneut Tabulatortaste dr\u00
 jshell.console.completion.all.completions = <Erneut Tabulatortaste dr\u00FCcken, um alle m\u00F6glichen Abschl\u00FCsse anzuzeigen>
 jshell.console.no.javadoc = <Keine Dokumentation gefunden>
 jshell.console.do.nothing = Nichts machen
-jshell.console.choice = Auswahl: 
+jshell.console.choice = Auswahl:\u0020
```
Comment (justin-curtis-lu) [inline src/jdk.jshell/share/classes/jdk/internal/jshell/tool/resources/l10n_de.properties:170]: This is correct.  However, the original l10n.properties did not get the change as intended.

l10n.properties had it implemented as 

`jshell.console.choice = Choice: \`

The trailing white space was enclosed with a trailing `\` (probably to signify the white space is intentional) and thus hidden from the script.

It should be changed to `jshell.console.choice = Choice:\u0020` for consistency

**Comment by magicus** [comment]
Bot-Keep-Alive-Ping

**Comment by magicus** [comment]
For the record, when I tried to revive this, it turned out that all changes proposed here were already implemented in master.

**Comment by justin-curtis-lu** [comment]
Hi @magicus, the trailing white space in all .properties files between the localized and un-localized versions was standardized in [8300719](https://bugs.openjdk.org/browse/JDK-8300719) as part of the L10n translations. The changes there likely included the proposed changes here as well.

For context, there was no removal of questionable trailing white space, only the standardization that the localized versions matched the original version. And similar to this patch, `/u0020` was used in replacement of ` `. We usually see that the external translation tool causes inconsistency in trailing white space, so we re-run our white space tool during each L10n translation drop.

**Pull request closed without merging**